### PR TITLE
Rem 438

### PR DESCRIPTION
**Pull request by Alinadimmokdad**
rt-fees-schedule/
rt-fees-schedule-map/

### CODE REVIEW

**Comment by Ghinwa-Arab** [comment]
image

origin currency mandatory

**Comment by Ghinwa-Arab** [comment]
image

after save info button still disable

**Comment by Ghinwa-Arab** [comment]
image
make sure transaction log returns data in both pages

**Comment by Ghinwa-Arab** [comment]
image

clear button not working

File: src/pages/rt-fees-schedule/index.js [hunk 1]
```diff
@@ -0,0 +1,124 @@
+import { useState, useContext } from 'react'
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/index.js:1]: remove state

File: src/pages/rt-fees-schedule/index.js [hunk 2]
```diff
@@ -0,0 +1,124 @@
...
+import { RemittanceOutwardsRepository } from 'src/repositories/RemittanceOutwardsRepository'
+import FeesSceduleWindow from './window/FeesSceduleWindow'
+
+const FeeSchedule = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  const { stack } = useWindow()
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50 } = options
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/index.js:26]: remove

File: src/pages/rt-fees-schedule/index.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+
+    return await getRequest({
+      extension: RemittanceOutwardsRepository.FreeSchedule.qry,
+      parameters: `_startAt=${_startAt}&_pageSize=${_pageSize}&filter=`
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/index.js:30]: we don't have pagination please remove pagination params (startAt,pageSize)

File: src/pages/rt-fees-schedule/index.js [hunk 4]
```diff
@@ -0,0 +1,124 @@
...
+  const add = () => {
+    openForm()
+  }
+
+  function openForm(recordId) {
+    stack({
+      Component: FeesSceduleWindow,
+      props: {
+        labels: _labels,
+        recordId: recordId ? recordId : null,
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/index.js:78]: no need for this `recordId ? recordId : null,` pass only recordId

File: src/pages/rt-fees-schedule/index.js [hunk 5]
```diff
@@ -0,0 +1,124 @@
...
+    })
+  }
+
+  const edit = obj => {
+    openForm(obj.recordId)
+  }
+
+  const del = async obj => {
+    await postRequest({
+      extension: RemittanceOutwardsRepository.FreeSchedule.del,
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/index.js:93]: add try catch

File: src/pages/rt-fees-schedule/index.js [hunk 6]
```diff
@@ -0,0 +1,124 @@
...
+      record: JSON.stringify(obj)
+    })
+    invalidate()
+    toast.success('Record Deleted Successfully')
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/index.js:97]: use platform labels

File: src/pages/rt-fees-schedule/window/FeesSceduleWindow.js [hunk 7]
```diff
@@ -0,0 +1,46 @@
+// ** Custom Imports
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/window/FeesSceduleWindow.js:1]: remove

File: src/pages/rt-fees-schedule/window/FeesSceduleWindow.js [hunk 8]
```diff
@@ -0,0 +1,46 @@
...
+import CustomTabPanel from 'src/components/Shared/CustomTabPanel'
+import { CustomTabs } from 'src/components/Shared/CustomTabs'
+import { useState } from 'react'
+import FeesSceduleForm from '../forms/FeesSceduleForm'
+import FeesDetailsForm from '../forms/FeesDetailsForm'
+
+const FeesSceduleWindow = ({ height, recordId, labels, maxAccess, expanded }) => {
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/window/FeesSceduleWindow.js:8]: remove expanded, not used anywhere

Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/window/FeesSceduleWindow.js:8]: is height necessary?? please remove height from the whole file it seems that serving no purpose

File: src/pages/rt-fees-schedule/window/FeesSceduleWindow.js [hunk 9]
```diff
@@ -0,0 +1,46 @@
...
+  const [activeTab, setActiveTab] = useState(0)
+
+  const editMode = !!recordId
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/window/FeesSceduleWindow.js:11]: remove editMode, u're passing it to tabs but it's not used there anywhere

File: src/pages/rt-fees-schedule/window/FeesSceduleWindow.js [hunk 10]
```diff
@@ -0,0 +1,46 @@
...
+          labels={labels}
+          setStore={setStore}
+          store={store}
+          editMode={editMode}
+          maxAccess={maxAccess}
+          onChange={onStrategiesChange}
+        />
+      </CustomTabPanel>
+      <CustomTabPanel height={height} index={1} value={activeTab}>
+        <FeesDetailsForm labels={labels} setStore={setStore} maxAccess={maxAccess} store={store} editMode={editMode} />
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/window/FeesSceduleWindow.js:39]: maxAccess and setStore not used inside FeesDetailsForm

File: src/pages/rt-fees-schedule/forms/FeesSceduleForm.js [hunk 11]
```diff
@@ -0,0 +1,145 @@
+// ** MUI Imports
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesSceduleForm.js]: remove

File: src/pages/rt-fees-schedule/forms/FeesSceduleForm.js [hunk 12]
```diff
@@ -0,0 +1,145 @@
...
+import { Grid } from '@mui/material'
+import { DataSets } from 'src/resources/DataSets'
+
+import * as yup from 'yup'
+import toast from 'react-hot-toast'
+
+// ** Custom Imports
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesSceduleForm.js]: remove

File: src/pages/rt-fees-schedule/forms/FeesSceduleForm.js [hunk 13]
```diff
@@ -0,0 +1,145 @@
...
+
+  const postGroups = async obj => {
+    const isNewRecord = !obj?.recordId
+    try {
+      const res = await postRequest({
+        extension: RemittanceOutwardsRepository.FreeSchedule.set,
+        record: JSON.stringify(obj)
+      })
+
+      const message = isNewRecord ? 'Record Added Successfully' : 'Record Edited Successfully'
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesSceduleForm.js:57]: use platform labels

File: src/pages/rt-fees-schedule/forms/FeesSceduleForm.js [hunk 14]
```diff
@@ -0,0 +1,145 @@
...
+      postGroups(values)
+    }
+  })
+
+  useEffect(() => {
+    onChange(formik.values)
+  }, [formik.values])
+
+  const postGroups = async obj => {
+    const isNewRecord = !obj?.recordId
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesSceduleForm.js:50]: remove const and use obj.recordId directly

File: src/pages/rt-fees-schedule/forms/FeesSceduleForm.js [hunk 15]
```diff
@@ -0,0 +1,145 @@
...
+      name: yup.string().required(),
+      reference: yup.string().required(),
+      originCurrency: yup.string().required()
+    }),
+    onSubmit: values => {
+      postGroups(values)
+    }
+  })
+
+  useEffect(() => {
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesSceduleForm.js:45]: what the purpose of useEffect?

File: src/pages/rt-fees-schedule/forms/FeesSceduleForm.js [hunk 16]
```diff
@@ -0,0 +1,145 @@
...
+    onChange(formik.values)
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesSceduleForm.js:46]: we care about recordId only, why u're updating setStore whenever formik values changes?

File: src/pages/rt-fees-schedule/forms/FeesSceduleForm.js [hunk 17]
```diff
@@ -0,0 +1,145 @@
...
+    if (recordId) {
+      getStrategyId(recordId)
+    }
+  }, [])
+
+  const getStrategyId = recordId => {
+    const defaultParams = `_recordId=${recordId}`
+    getRequest({
+      extension: RemittanceOutwardsRepository.FreeSchedule.get,
+      parameters: defaultParams
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesSceduleForm.js:84]: remove defaultParams and pass ``_recordId=${recordId}`` directly here

File: src/pages/rt-fees-schedule/forms/FeesDetailsForm.js [hunk 18]
```diff
@@ -0,0 +1,156 @@
+import { Grid, Box, Checkbox } from '@mui/material'
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesDetailsForm.js:1]: remove
import { Grid, Box, Checkbox } from '@mui/material'
ResourceIds
useState
yup
they are not used

File: src/pages/rt-fees-schedule/forms/FeesDetailsForm.js [hunk 19]
```diff
@@ -0,0 +1,156 @@
...
+import * as yup from 'yup'
+import toast from 'react-hot-toast'
+import { RequestsContext } from 'src/providers/RequestsContext'
+import { ResourceIds } from 'src/resources/ResourceIds'
+import { VertLayout } from 'src/components/Shared/Layouts/VertLayout'
+import { Grow } from 'src/components/Shared/Layouts/Grow'
+import { DataSets } from 'src/resources/DataSets'
+import { RemittanceOutwardsRepository } from 'src/repositories/RemittanceOutwardsRepository'
+
+const FeesDetailsForm = ({ store, labels, editMode, maxAccess }) => {
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesDetailsForm.js:13]: remove  editMode, maxAccess
they are not used

File: src/pages/rt-fees-schedule/forms/FeesDetailsForm.js [hunk 20]
```diff
@@ -0,0 +1,156 @@
...
+  const { recordId: pId } = store
+  const { getRequest, postRequest } = useContext(RequestsContext)
+
+  useEffect(() => {
+    getGridData(pId)
+  }, [pId])
+
+  function getGridData(scheduleId) {
+    formik.setValues({
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesDetailsForm.js:22]: what the purpose of this? if u already set formik initial values

File: src/pages/rt-fees-schedule/forms/FeesDetailsForm.js [hunk 21]
```diff
@@ -0,0 +1,156 @@
...
+        }
+      ]
+    })
+
+    if (scheduleId) {
+      var parameters = `_scheduleId=${scheduleId}`
+      getRequest({
+        extension: RemittanceOutwardsRepository.FreeScheduleDetail.qry,
+
+        parameters: parameters
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesDetailsForm.js:42]: pass ` `_scheduleId=${scheduleId}`` directly

File: src/pages/rt-fees-schedule/forms/FeesDetailsForm.js [hunk 22]
```diff
@@ -0,0 +1,156 @@
...
+        .then(res => {
+          const result = res.list
+          console.log(result, 'resss')
+
+          const processedData = result.map((item, index) => ({
+            ...item,
+            id: index + 1,
+            seqNo: index + 1
+          }))
+          console.log(processedData, 'prooo')
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesDetailsForm.js]: remove

File: src/pages/rt-fees-schedule/forms/FeesDetailsForm.js [hunk 23]
```diff
@@ -0,0 +1,156 @@
...
+    if (scheduleId) {
+      var parameters = `_scheduleId=${scheduleId}`
+      getRequest({
+        extension: RemittanceOutwardsRepository.FreeScheduleDetail.qry,
+
+        parameters: parameters
+      })
+        .then(res => {
+          const result = res.list
+          console.log(result, 'resss')
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesDetailsForm.js]: remove

File: src/pages/rt-fees-schedule/forms/FeesDetailsForm.js [hunk 24]
```diff
@@ -0,0 +1,156 @@
...
+
+          const processedData = result.map((item, index) => ({
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesDetailsForm.js:47]: replace result by res.list , and remove const above

File: src/pages/rt-fees-schedule/forms/FeesDetailsForm.js [hunk 25]
```diff
@@ -0,0 +1,156 @@
...
+      scheduleId: pId,
+      items: res
+    }
+
+    postRequest({
+      extension: RemittanceOutwardsRepository.FreeScheduleDetail.set2,
+      record: JSON.stringify(data)
+    })
+      .then(res => {
+        toast.success('Record Successfully')
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/forms/FeesDetailsForm.js:131]: use platform labels

File: src/pages/rt-fees-schedule-map/index.js [hunk 26]
```diff
@@ -0,0 +1,151 @@
...
+const FreeScheduleMap = () => {
+  const { getRequest, postRequest } = useContext(RequestsContext)
+  const { platformLabels } = useContext(ControlContext)
+
+  const { stack } = useWindow()
+
+  async function fetchGridData(options = {}) {
+    const { _startAt = 0, _pageSize = 50 } = options
+
+    const response = await getRequest({
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule-map/index.js:26]: remove pagination params

File: src/pages/rt-fees-schedule-map/index.js [hunk 27]
```diff
@@ -0,0 +1,151 @@
...
+    })
+
+    return { ...response, _startAt: _startAt }
+  }
+
+  const {
+    query: { data },
+    labels: _labels,
+    paginationParameters,
+    refetch,
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule-map/index.js:39]: add invalidate here and remove one below

File: src/pages/rt-fees-schedule-map/index.js [hunk 28]
```diff
@@ -0,0 +1,151 @@
...
+
+  const add = () => {
+    openForm()
+  }
+
+  const edit = obj => {
+    openForm(obj)
+  }
+
+  const del = async obj => {
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule-map/index.js:97]: add try catch

File: src/pages/rt-fees-schedule-map/index.js [hunk 29]
```diff
@@ -0,0 +1,151 @@
...
+          columns={columns}
+          gridData={data}
+          rowId={['corId , functionId , currencyId ,countryId,dispersalMode']}
+          onEdit={edit}
+          onDelete={del}
+          isLoading={false}
+          pageSize={50}
+          refetch={refetch}
+          paginationParameters={paginationParameters}
+          paginationType='api'
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule-map/index.js:143]: replace to client

File: src/pages/rt-fees-schedule/index.js [hunk 30]
```diff
@@ -0,0 +1,124 @@
...
+          columns={columns}
+          gridData={data}
+          rowId={['recordId']}
+          onEdit={edit}
+          onDelete={del}
+          isLoading={false}
+          pageSize={50}
+          refetch={refetch}
+          paginationParameters={paginationParameters}
+          paginationType='api'
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule/index.js:116]: replace type to client

File: src/pages/rt-fees-schedule-map/FreeScheduleMapForm.js [hunk 31]
```diff
@@ -0,0 +1,278 @@
+import { Grid } from '@mui/material'
+import { useContext, useEffect, useState } from 'react'
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule-map/FreeScheduleMapForm.js]: remove useState

File: src/pages/rt-fees-schedule-map/FreeScheduleMapForm.js [hunk 32]
```diff
@@ -0,0 +1,278 @@
...
+      corId: '',
+
+      currencyId: '',
+      countryId: '',
+      dispersalMode: '',
+      functionId: '',
+      scheduleId: '',
+      functionName: '',
+
+      ...record
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule-map/FreeScheduleMapForm.js]: what the purpose of   ...record?

File: src/pages/rt-fees-schedule-map/FreeScheduleMapForm.js [hunk 33]
```diff
@@ -0,0 +1,278 @@
...
+      scheduleId: yup.string().required(),
+      dispersalMode: yup.string().required(),
+      countryId: yup.string().required()
+    }),
+
+    //   onSubmit: async obj => {
+    //     const recordId = obj.recordId
+
+    //     const response = await postRequest({
+    //       extension: RemittanceOutwardsRepository.FreeScheduleMap.set,
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule-map/FreeScheduleMapForm.js]: remove

File: src/pages/rt-fees-schedule-map/FreeScheduleMapForm.js [hunk 34]
```diff
@@ -0,0 +1,278 @@
...
+
+      await postRequest({
+        extension: RemittanceOutwardsRepository.FreeScheduleMap.set,
+        record: JSON.stringify(obj)
+      })
+
+      if (!currencyId && !corId && !dispersalMode && !functionId && !countryId) {
+        toast.success(platformLabels.Added)
+      } else toast.success(platformLabels.Edited)
+      formik.setValues({
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule-map/FreeScheduleMapForm.js]: use formik.setFieldValue , no need for destructing

File: src/pages/rt-fees-schedule/forms/FeesSceduleForm.js (hunk 16 above)
Comment (Alinadimmokdad) [reply src/pages/rt-fees-schedule/forms/FeesSceduleForm.js:46]: what if we want to add a store

File: src/pages/rt-fees-schedule-map/FreeScheduleMapForm.js [hunk 35]
```diff
@@ -0,0 +1,278 @@
...
+              String(res.record.dispersalMode * 10) +
+              String(res.record.functionId) +
+              String(res.record.countryId * 100)
+          })
+        }
+      } catch (exception) {}
+    })()
+  }, [])
+
+  console.log(formik.values.functionId)
```
Comment (Ghinwa-Arab) [inline src/pages/rt-fees-schedule-map/FreeScheduleMapForm.js]: remove

File: src/pages/rt-fees-schedule/forms/FeesDetailsForm.js (hunk 20 above)
Comment (Alinadimmokdad) [reply src/pages/rt-fees-schedule/forms/FeesDetailsForm.js:22]: new Row

File: src/pages/rt-fees-schedule-map/index.js [hunk 36]
```diff
@@ -0,0 +1,151 @@
...
+  }
+
+  function openForm(record) {
+    stack({
+      Component: FreeScheduleMapForm,
+      props: {
+        labels: _labels,
+        record,
+        maxAccess: access,
+        recordId: record
```
[the remark this answers is not in the document]
Comment (Alinadimmokdad) [reply src/pages/rt-fees-schedule-map/index.js:113]: no we need the record id from index

**Pull request merged**